### PR TITLE
Small documentation fixes

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #381

Fixing strings that aren't rendered correctly and add the ones that are missing from documentation (e.g. S3 DataPipes).

Differential Revision: [D36076937](https://our.internmc.facebook.com/intern/diff/D36076937)

### CODE REVIEW

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36076937).

File: docs/source/conf.py [hunk 1]
```diff
@@ -108,6 +110,16 @@
...
+    "torchdata.datapipes.map.util.cacheholder.T_co": "T_co",
+    "Sequence[torchdata.datapipes.map.util.unzipper.T]": "Sequence[T]",
+    "torchdata.datapipes.iter.util.samplemultiplexer.T_co": "T_co",
+    "torchdata.datapipes.iter.util.indexadder.K": "K",
+    "torchdata.datapipes.iter.util.unzipper.T": "T",
+    "torch.utils.data.datapipes.iter.grouping.T_co": "T_co",
+    "torchdata.datapipes.iter.util.dataframemaker.T_co": "T_co",
+    "torchdata.datapipes.iter.util.cacheholder.T_co": "T_co",
+    "torchdata.datapipes.iter.util.header.T_co": "T_co",
+    "<class 'torch.utils.data.datapipes.datapipe.DataChunk'>": "torch.utils.data.datapipes.datapipe.DataChunk",
```
Comment (ejguan) [inline docs/source/conf.py:122]: Could we remove `DataChunk` from the doc? This is not supposed to be exposed to users IMHO.

File: torchdata/datapipes/iter/util/converter.py [hunk 2]
```diff
@@ -26,6 +26,8 @@ class IterToMapConverterMapDataPipe(MapDataPipe):
     with exactly two objects. The first object of each item becomes a key in
     the new dictionary, and the second object the corresponding value.
 
+    For the opposite converter, use ``MapToIterConverter``.
```
Comment (ejguan) [inline torchdata/datapipes/iter/util/converter.py:29]: IIRC, we can do something like `:class:torchdata.datapipes.map.MapToIterConverter`

File: torchdata/datapipes/map/util/converter.py [hunk 3]
```diff
@@ -15,6 +15,8 @@ class MapToIterConverterIterDataPipe(IterDataPipe):
     Convert a ``MapDataPipe`` to an ``IterDataPipe`` (functional name: ``to_iter_datapipe``). It uses ``indices`` to
     iterate through the ``MapDataPipe``, defaults to ``range(len(mapdatapipe))`` if not given.
 
+    For the opposite converter, use ``IterToMapConverter``.
```
Comment (ejguan) [inline torchdata/datapipes/map/util/converter.py:18]: Ditto

**Review by ejguan — APPROVED** [review]
Thank you, LGTM

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36076937).

File: torchdata/datapipes/iter/util/converter.py [hunk 4]
```diff
@@ -26,13 +26,22 @@ class IterToMapConverterMapDataPipe(MapDataPipe):
     with exactly two objects. The first object of each item becomes a key in
     the new dictionary, and the second object the corresponding value.
 
+    For the opposite converter, use :class:`.MapToIterConverter`.
```
Comment (ejguan) [inline torchdata/datapipes/iter/util/converter.py:29]: Wow. Have you checked this is gonna work?

Comment (NivekT) [reply torchdata/datapipes/iter/util/converter.py:29]: I did and it does!

Comment (ejguan) [reply torchdata/datapipes/iter/util/converter.py:29]: Thank you for verification!

Comment (NivekT) [reply torchdata/datapipes/iter/util/converter.py:29]: Screen Shot 2022-05-03 at 2 17 23 PM

**Pull request closed without merging**